jobs/config.json

```diff
@@ -8519,6 +8519,27 @@
       "sig-cluster-lifecycle"
     ]
   },
+  "ci-kubernetes-e2e-kubeadm-gce-1-6-on-1-7-upgrade-cluster-init": {
+    "args": [
+      "--cluster=",
+      "--deployment=kubernetes-anywhere",
+      "--env-file=jobs/platform/gce.env",
+      "--extract=ci/latest-1.6",
+      "--gcp-zone=us-central1-f",
+      "--kubeadm=ci",
+      "--kubernetes-anywhere-kubernetes-version=latest-1.6",
+      "--kubernetes-anywhere-upgrade-provider=kubeadm",
+      "--kubernetes-anywhere-kubeadm-upgrade-method=init",
+      "--provider=kubernetes-anywhere",
+      "--test_args=--ginkgo.focus=\\[Conformance\\] --minStartupPods=8",
+      "--upgrade_args=--ginkgo.focus=\\[Feature:ClusterUpgrade\\] --upgrade-target=ci/latest-1.7",
```

Comment:
So that the upgrade tests use kubernetes anywhere instead of the kubernetes upgrade script, we need to add a fork for provider kubernetes anywhere that will call the kuberetes anywhere upgrade-master https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/nodes_util.go#L43

Reply:
Ah, I see. Looking at [`MasterUpgrade `](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/nodes_util.go#L43), its switch is based on the `--provider` flag. Since many other helper functions rely on this flag being `gce`, is it safe to change the value to `kubernetes-anywhere` for this job? If not, do we need to create a new `--upgrade-provider` flag maybe?

Reply:
When I skimmed the code, I did not see the Master Upgrade Test (https://github.com/kubernetes/kubernetes/blob/ad0d36f0f0f5cf02ca2b032b386b5b1f6c313cf5/test/e2e/lifecycle/cluster_upgrade.go#L79) using any helper functions (other than MasterUpgrade) that depended on the provider flag. Have you found some?

All the other end to end kubeadm tests already set the provider as kubernetes-anywhere. If we want to switch that back to gce, we would need to sanity check for logic that depends on the provider being kubernetes-anywhere. 
@pipejakob - Do you know of any logic that requires the provider to be kubernetes-anywhere for the kubeadm e2e jobs?

A third option to consider is the deployer flag. The deployer is what creates and deletes the cluster. It would make sense that the style of creation and deletion would control the style of upgrade.